### PR TITLE
feat: filter to show verified or unverified users

### LINKED ISSUE #3330: Add ability to filter auth table by able to show only verified or unverified email of auth user
Once there is too many user registered its hard to track user that are verified and not yet verified

### PR DESCRIPTION
**Pull request by Hallidayo**
## What kind of change does this PR introduce?

Studio Feature
Fixed https://github.com/supabase/supabase/issues/3330

## What is the current behavior?

At the moment in Supabase Studio Auth their is only a search field for an email and no filter to show verified/unverified users:

Screenshot 2022-06-06 at 09 35 34

## What is the new behavior?

A new filter has been added so that the list can be filtered to show all, verified and unverified users in the list:

Screenshot 2022-06-06 at 09 38 00

https://user-images.githubusercontent.com/22655069/172126476-01d23439-12e9-4324-9250-19609ec9d7cb.mov

## Additional context

This is linked to issue #3330

@MildTomato @phamhieu @joshenlim

### CODE REVIEW

**Comment by saltcod** [comment]
Hey @Hallidayo ! This is great, thank you! 

I think a plain Listbox makes sense for this one instead of a Select, and we can probably skip the icon.  We should also add some margin between the search box and this listbox. 

Last, the filtering isn't working for me locally. It sends requests, but it doesn't filter the list. I've got 1 verified and 1 unverified in my instance. Any ideas?

Here's the two requests:

CleanShot 2022-06-06 at 11 49 37@2x

**Comment by Hallidayo** [comment]
Hi @saltcod, Thank you for checking over this.

I've made the changes you mentioned and now the filter is looking like so:
Screenshot 2022-06-06 at 16 06 32

So now the `<Listbox/>` component is being used and also some margin has been added.

When you are filtering locally do you get a spinner appearing as if a request has been made?

**Comment by saltcod** [comment]
@Hallidayo I do see the spinner! And requests are being made, but I get the same results back.

**Comment by Hallidayo** [comment]
@saltcod - The verified users is running the following query if no search word is being used:

`SELECT * from auth.users WHERE (email_confirmed_at IS NOT NULL or phone_confirmed_at IS NOT NULL) ORDER BY created_at DESC LIMIT ? OFFSET ?`

Whilst un-verified is doing the following:

`SELECT * from auth.users WHERE (email_confirmed_at IS NULL AND phone_confirmed_at IS NULL) ORDER BY created_at DESC LIMIT ? OFFSET ?`

Are the values for `email_confirmed_at` and `phone_confirmed_at` in your `auth.users` table set to `NULL`? Whilst testing I had to set values for one of these fields in in the `auth.users` table so I could test the different scenarios

**Comment by saltcod** [comment]
Hey @Hallidayo Just added a new issue to our internal infrastructure repo. We need to add support for this query param there as well before we can merge this one. I'll update this issue when that one gets addressed. Thanks for the help on this one so far!

**Comment by Hallidayo** [comment]
@saltcod - Thats great, thank you for keeping me updated.

File: studio/pages/api/auth/[ref]/users.ts [hunk 1]
```diff
@@ -25,23 +25,69 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
...
+        'SELECT * from auth.users WHERE (email_confirmed_at IS NULL AND phone_confirmed_at IS NULL) ORDER BY created_at DESC LIMIT ? OFFSET ?;',
+        [limitInt, offsetInt]
+      )
+    }
+  }
+
+  if (hasValidKeywords && hasVerifiedValue) {
+    if (verified === 'verified') {
+      queryCount = SqlString.format(
+        'SELECT count(*)  from auth.users WHERE (email_confirmed_at IS NOT NULL or phone_confirmed_at IS NOT NULL) email ilike ?;',
```
Comment (phamhieu) [inline studio/pages/api/auth/[ref]/users.ts:70]: Missing `AND`

Comment (Hallidayo) [reply studio/pages/api/auth/[ref]/users.ts]: @phamhieu - Just added a commit with the change.

**Comment by saltcod** [comment]
Hey @Hallidayo ! I'm getting a 404 on this request:

CleanShot 2022-08-31 at 11 48 18@2x

http://localhost:8080/platform/auth/[ref]/users?limit=10&offset=0&keywords=&verified=

**Comment by Hallidayo** [comment]
@saltcod - Just ran locally and I'm getting the following:

https://user-images.githubusercontent.com/22655069/187709927-9efe5148-8c5b-418c-9abd-bcac095ca2d8.mov

All network requests are going through fine as well with no 404, On the request whats does the Preview/Repsonse look like?

**Comment by saltcod** [comment]
Weird. The response is:
CleanShot 2022-08-31 at 12 42 12@2x

File: studio/pages/api/auth/[ref]/users.ts (hunk 1 above)
Comment (saltcod) [reply studio/pages/api/auth/[ref]/users.ts]: @phamhieu is this querying for you? I get a 404 on the users request.

**Comment by pucho** [comment]
I'm currently having this issue when trying to get this to work

Screen Shot 2022-09-06 at 12 57 54

**Comment by Hallidayo** [comment]
Do you think it's something to do with the `verified` query param that is being used, Shall I change `verified` to `verifiedValue`?

**Review by pucho — CHANGES_REQUESTED** [review]

**Comment by Hallidayo** [comment]
@pucho  - just done a refactor and pushed a commit.

**Review by pucho — APPROVED** [review]

**Comment by pucho** [comment]
@Hallidayo there seems to be a weird issue with github picking up a conflict that does not exist. Do you mind creating a new branch from a fresh master pull, add these changes and I can approve again?

Sorry for the inconvenience, let me know on Discord if you need anything. Thanks!!!

**Comment by Hallidayo** [comment]
@pucho yeah sure no problem

**Comment by pucho** [comment]
Closed in favor of https://github.com/supabase/supabase/pull/8924

**Pull request closed without merging**